src/core/public/http/types.ts

```diff
@@ -206,12 +206,19 @@ export interface HttpRequestInit {
 
 /** @public */
 export interface HttpFetchQuery {
-  [key: string]:
-    | string
-    | number
-    | boolean
-    | undefined
-    | Array<string | number | boolean | undefined>;
+  /**
+   * TypeScript note: Technically we should use this interface instead, but @types/node uses the below stricter
+   * definition, so to avoid TypeScript errors, we'll restrict our version.
+   *
+   * [key: string]:
+   *   | string
+   *   | number
+   *   | boolean
+   *   | Array<string | number | boolean>
+   *   | undefined
+   *   | null;
+   */
+  [key: string]: string | number | boolean | string[] | number[] | boolean[] | undefined | null;
```

Comment:
@oatkiller I had to slightly revert part of #59340 because of a conflict with a new version of `@types/node`:

In Node.js 10, the first argument to `querystring.stringify` was defined as:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/47bbc5e136c92275fbb51b6458e5b48c06739f81/types/node/v10/querystring.d.ts#L13

In Node.js 12, the type is more strict:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9ea38ba16ce74f125367446a1697c67b90e8ca42/types/node/v12/querystring.d.ts#L14

This interface conflicts in Node.js 12 because `Array<string | number | boolean | undefined>` is more loose compared to `string[] | number[] | boolean[]` (at least I assume that's the reason for the error I'm getting).

It's complaining about the `query` param here:
https://github.com/elastic/kibana/blob/1246dc35a3730234609b02108d4c57344477ac21/src/core/public/http/fetch.ts#L137-L140

Are you ok with this change and/or do you have any suggestion of how to work around this issue in a better way?

Reply:
@watson thanks for the changes. This seems perfect to me. My only intention w/ my prior change was to allow passing arrays. It isn't important that the arrays can contain a mix of types.

Reply:
> It isn't important that the arrays can contain a mix of types

Seems minor. Maybe it's even better that way.

The new `[key: string]: string | number | boolean | string[] | number[] | boolean[] | undefined | null` also added `null` as an allowed value though. Minor I guess, but is there a reason for that, as removing it should keep compatibility with the `node` type?

Reply:
@pgayvallet I simply inlined the types from `@types/node` which include both `undefined` and `null`. What's the reason we would not simply align with `@types/node`?

Reply:
Main reason would be because we may be using this `HttpFetchQuery` type for other usages that `url.format` and node APIs and therefor have undocumented / unpredictable behavior with the introduced `null` value that wasn't allowed before. 

But it's effectively very unlikely, so I'm ok aligning with `node` types.